### PR TITLE
Templatedecoratorrework

### PR DESCRIPTION
**Pull request by khituras**
Integrated the new `JsonTemplateQueryDecorator` that works with templates which adhere to the JSON syntax. Please check if I forgot an important functionality. All features are described in the documentation of the README.md file.

### CODE REVIEW

File: src/test/java/at/medunigraz/imi/bst/TestTopic.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    @QueryDescriptionAttribute
+    private List<String> stopFilteredTermList;
+    @QueryDescriptionAttribute
+    private String[] stopFilteredTermArray;
+    @QueryDescriptionAttribute
+    private double weight;
+    @QueryDescriptionAttribute
+    private List<String[]> friends;
+
+    public TestTopic withFriends(String[]... friends) {
```
Comment (michelole) [inline src/test/java/at/medunigraz/imi/bst/TestTopic.java:21]: Friends?

File: src/test/resources/test-subtemplates-truejson/friendsBoolDoubleForIndexIn.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "bool": {
+    "must": [
+      "${FOR INDEX IN friends[] REPEAT friendsMatch.json}"
```
Comment (michelole) [inline src/test/resources/test-subtemplates-truejson/friendsBoolDoubleForIndexIn.json:4]: Wow. Is this a standard?

File: src/test/java/at/medunigraz/imi/bst/TestTopic.java (hunk 1 above)
Comment (khituras) [reply src/test/java/at/medunigraz/imi/bst/TestTopic.java:21]: Sure. Everyone should have some! Why not also our TestTopic ;-)

**Comment by michelole** [comment]
Shouldn't we update all old templates?

**Comment by khituras** [comment]
Ideally, yes. But we would need to test them all. Seems like quite some work. On the other hand, the old decorators still work and thus, so do the old templates. You think it would be important to update them and delete the old decorators all the same?

**Comment by michelole** [comment]
I'd say it could be confusing to have two competing "template engines".

Maybe that could be a follow up work if it's too much to do now --- we could then also clean up outdated templates/decorators.

**Comment by khituras** [comment]
You're right, it would be confusing and I'm not a fan myself. Feel free to start the process. I'd join but I won't do it alone.

**Pull request merged**